internal/sharedresource/sharedresource.go

```diff
@@ -0,0 +1,80 @@
+package sharedresource
+
+import (
+	"github.com/go-logr/logr"
+	"github.com/manifestival/manifestival"
+	openshiftv1alpha1 "github.com/redhat-openshift-builds/operator/api/v1alpha1"
+	"github.com/redhat-openshift-builds/operator/internal/common"
+	"k8s.io/apimachinery/pkg/api/errors"
+)
+
+// SharedResource type defines methods to Get, Create v1alpha1.SharedResource resource
+type SharedResource struct {
+	Logger   logr.Logger
+	Manifest manifestival.Manifest
+	State    openshiftv1alpha1.State
+}
+
+// New creates new instance of SharedResource type
+func New(manifest manifestival.Manifest) *SharedResource {
+	return &SharedResource{
+		Manifest: manifest,
+	}
+}
+
+// UpdateSharedResource transforms the manifests, and applies or deletes them based on SharedResource.State.
+func (sr *SharedResource) UpdateSharedResources(owner *openshiftv1alpha1.OpenShiftBuild) error {
+	logger := sr.Logger.WithValues("name", owner.Name)
+	sr.State = owner.Spec.SharedResource.State
+	images := common.ImagesFromEnv(common.SharedResourceImagePrefix)
+
+	// Applying transformers
+	transformerfuncs := []manifestival.Transformer{}
+	transformerfuncs = append(transformerfuncs, manifestival.InjectOwner(owner))
+	transformerfuncs = append(transformerfuncs, manifestival.InjectNamespace(common.OpenShiftBuildNamespaceName))
+	transformerfuncs = append(transformerfuncs, common.InjectContainerImages(images))
```

Comment:
Along the lines of prior comments - we won't need this if we just add the pull specs of the container images directly, and let Konflux take care of the rest.